### PR TITLE
Fix computation error on disclosure (take 2 🙈)

### PR DESCRIPTION
**Pull request by alex-ju**
### :pushpin: Summary

Fix computation error on disclosure utility component.

### :hammer_and_wrench: Detailed description

In https://github.com/hashicorp/design-system/pull/456 I wrongly identified the issue – the solution made the `cloud-ui` tests pass, but only sometimes. After managing to replicate the failing tests locally and testing with https://github.com/hashicorp/cloud-ui/pull/3098 it turns out the racing happens between the `@action` functions (`onFocusOut`, `onKeyUp`) triggering the same `notifyPropertyChange` via `close` in a short amount of time (even shorter in automated tests). I tried many options to prevent this error from happening in tests without avail. The only solution I could find that prevents tests in `cloud-ui` from failing intermittently is to move what happens in the `close` function to the next runloop – hence I'm proposing this to unblock the adoption of this component until a wiser solution may be determined.

***

### Preview
Failing/ﬂaky test in HCP consul

#### Before
Screenshot 2022-07-18 at 18 14 40

#### After
Screenshot 2022-07-18 at 18 11 33

### 👀 How to review

👉 Review by files changed

Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: packages/components/addon/components/hds/disclosure/index.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
+import { later } from '@ember/runloop';
 
 export default class HdsDisclosureComponent extends Component {
   @tracked isOpen; // notice: if in the future we need to add a "@isOpen" prop to control the status from outside (eg to have the Disclosure opened on render) just add  "this.args.isOpen" here to initalize the variable
   @tracked toggleRef;
-  @tracked isToggleClicked;
```
Comment (alex-ju) [inline packages/components/addon/components/hds/disclosure/index.js:8]: This was a leftover from refactoring – not used anywhere across the repo.

**Review by didoo — APPROVED** [review]
I don't see anything on my side, but I am not an Ember expert, so I would suggest to ask an extra couple of people with strong Ember background for review (or even better, post the PR in the #tech-frontend Slack channel for second opinions).

File: packages/components/addon/components/hds/disclosure/index.js [hunk 2]
```diff
@@ -48,12 +43,19 @@ export default class HdsDisclosureComponent extends Component {
...
+      function () {
+        if (this.isOpen) {
+          this.isOpen = false;
+        }
+        // we call the "onClose" callback if it exists (and is a function)
+        if (this.args.onClose && typeof this.args.onClose === 'function') {
+          this.args.onClose();
+        }
+      },
+      100
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply packages/components/addon/components/hds/disclosure/index.js]: yes, I thought about the same thing, and we don't actually need any milliseconds to move it to the next tick. just updated now to remove this.

File: packages/components/addon/components/hds/disclosure/index.js [hunk 3]
```diff
@@ -48,12 +43,12 @@ export default class HdsDisclosureComponent extends Component {
 
   @action
   close() {
-    if (this.isOpen) {
+    schedule('afterRender', () => {
```
Comment (didoo) [inline packages/components/addon/components/hds/disclosure/index.js:46]: @alex-ju missing a comment here why this is necessary (not sure why you have removed it)

**Review by didoo — APPROVED** [review]
OK for me

File: packages/components/addon/components/hds/disclosure/index.js (hunk 3 above)
Comment (alex-ju) [reply packages/components/addon/components/hds/disclosure/index.js:46]: good spot, I'll add a new one

**Pull request merged**